typescript/schedules.md

```diff
@@ -0,0 +1,1257 @@
+import { WorkflowHandle, WorkflowStartOptions } from './workflow-client';
+
+import type { NonNegativeInteger, RequireAtLeastOne } from 'type-fest';
+
+// TODO is a non-generic NonNegativeInteger possible? The ones below error due to no type argument
+export type PositiveInteger = Exclude<NonNegativeInteger, 0>;
+
+/**
+ * @format number of milliseconds or {@link https://www.npmjs.com/package/ms | ms-formatted string}
+ */
+export type Milliseconds = string | NonNegativeInteger;
+
+/**
+ * @format number of seconds or {@link https://www.npmjs.com/package/ms | ms-formatted string}
+ */
+export type Seconds = string | NonNegativeInteger;
+
+export interface UpdateScheduleOptions {
+  /**
+   * @default TODO
+   */
+  retry?: RetryPolicy;
+}
+
+export interface Backfill {
+  /** Time range to evaluate Schedule in. */
+  startAt: Date;
+  endAt: Date;
+
+  /** Override Overlap Policy for this request. */
+  overlapPolicyOverride?: ScheduleOverlapPolicy;
+}
+
+/**
+ * Handle to a single Schedule
+ */
+export interface ScheduleHandle {
+  /**
+   * This Schedule's identifier
+   */
+  readonly id: string;
+
+  /**
+   * Update the Schedule
+   */
+  update(mutateFn: (schedule: ScheduleDescription) => void | false, options?: UpdateScheduleOptions): Promise<void>;
+
+  /**
+   * Delete the Schedule
+   */
+  delete(): Promise<void>;
+
+  /**
+   * Trigger an Action to be taken immediately
+   */
+  trigger(): Promise<void>;
+
+  /**
+   * Run though the specified time period(s) and take Actions as if that time passed by right now, all at once. The
+   * Overlap Policy can be overridden for the scope of the Backfill.
+   */
+  backfill(options: Backfill | Backfill[]): Promise<void>;
+
+  /**
+   * Pause the Schedule
+   */
+  pause(note?: string): Promise<void>;
+
+  /**
+   * Unpause the Schedule
+   */
+  unpause(note?: string): Promise<void>;
+
+  /**
+   * Fetch the Schedule's description from the Server
+   */
+  describe(): Promise<ScheduleDescription>;
+
+  /**
+   * Get a handle to the most recent Action started
+   */
+  lastAction<T extends ScheduleActionType>(): Promise<HandleFor<T>>;
+
+  /**
+   * Readonly accessor to the underlying ScheduleClient
+   */
+  readonly client: ScheduleClient;
+}
+
+export interface Schedule {
+  /**
+   * Schedule Id
+   *
+   * We recommend using a meaningful business identifier.
+   */
+  id: string;
+
+  /**
+   * Additional non-indexed information attached to the Schedule. The values can be anything that is
+   * serializable by the {@link DataConverter}.
+   */
+  memo?: Record<string, any>;
+
+  /**
+   * Additional indexed information attached to the Schedule. More info:
+   * https://docs.temporal.io/docs/typescript/search-attributes
+   *
+   * Values are always converted using {@link JsonPayloadConverter}, even when a custom Data Converter is provided.
+   */
+  searchAttributes: SearchAttributes;
+
+  // TODO flatten this?
+  info: ScheduleInfo;
```

Comment:
Yes, if possible.
Ideally the list / describe response are as similar as possible to the create request.

Reply:
flattened